src/chocolatey/infrastructure.app/services/PowershellService.cs

```diff
@@ -403,22 +405,29 @@ public void prepare_powershell_environment(IPackage package, ChocolateyConfigura
 
             // we only want to pass the following args to packages that would apply. 
             // like choco install git -params '' should pass those params to git.install, 
-            // but not another package
-            if (!package_is_a_dependency_not_a_virtual(configuration, package.Id))
+            // but not another package unless the switch apply-install-arguments-to-dependencies is used
+            if (!package_is_a_dependency_not_a_virtual(configuration, package.Id) || configuration.ApplyInstallArgumentsToDependencies)
             {
-                this.Log().Debug(ChocolateyLoggers.Verbose, "Setting installer args and package parameters for {0}".format_with(package.Id));
+                this.Log().Debug(ChocolateyLoggers.Verbose, "Setting installer args for {0}".format_with(package.Id));
                 Environment.SetEnvironmentVariable("installArguments", configuration.InstallArguments);
                 Environment.SetEnvironmentVariable("installerArguments", configuration.InstallArguments);
                 Environment.SetEnvironmentVariable("chocolateyInstallArguments", configuration.InstallArguments);
-                Environment.SetEnvironmentVariable("packageParameters", configuration.PackageParameters);
-                Environment.SetEnvironmentVariable("chocolateyPackageParameters", configuration.PackageParameters);
 
                 if (configuration.OverrideArguments)
                 {
                     Environment.SetEnvironmentVariable("chocolateyInstallOverride", "true");
                 }
             }
 
+            // we only want to pass package parameters to packages that would apply. 
+            // but not another package unless the switch apply-package-parameters-to-dependencies is used
+            if (!package_is_a_dependency_not_a_virtual(configuration, package.Id) || configuration.ApplyPackageParametersToDependencies)
+            {
+                this.Log().Debug(ChocolateyLoggers.Verbose, "Setting package parameters for {0}".format_with(package.Id));
+                Environment.SetEnvironmentVariable("packageParameters", configuration.PackageParameters);
+                Environment.SetEnvironmentVariable("chocolateyPackageParameters", configuration.PackageParameters);
+            }
+

```

Comment:
Not a question, just a statement mostly. Something for me to consider later when pulling this in.